gobblin-hive-registration/src/main/java/org/apache/hadoop/hive/ql/io/orc/TypeDescriptionToObjectInspectorUtil.java

```diff
@@ -0,0 +1,443 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hadoop.hive.ql.io.orc;
+
+import java.util.ArrayList;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import org.apache.hadoop.hive.common.type.HiveDecimal;
+import org.apache.hadoop.hive.serde2.objectinspector.ListObjectInspector;
+import org.apache.hadoop.hive.serde2.objectinspector.MapObjectInspector;
+import org.apache.hadoop.hive.serde2.objectinspector.ObjectInspector;
+import org.apache.hadoop.hive.serde2.objectinspector.SettableListObjectInspector;
+import org.apache.hadoop.hive.serde2.objectinspector.SettableMapObjectInspector;
+import org.apache.hadoop.hive.serde2.objectinspector.SettableStructObjectInspector;
+import org.apache.hadoop.hive.serde2.objectinspector.StructField;
+import org.apache.hadoop.hive.serde2.objectinspector.UnionObjectInspector;
+import org.apache.hadoop.hive.serde2.objectinspector.primitive.PrimitiveObjectInspectorFactory;
+import org.apache.hadoop.hive.serde2.typeinfo.TypeInfoFactory;
+import org.apache.orc.OrcProto;
+import org.apache.orc.OrcUtils;
+import org.apache.orc.TypeDescription;
+
+
+public class TypeDescriptionToObjectInspectorUtil {
+  public static ObjectInspector getObjectInspector(TypeDescription orcSchema) {
+    return createObjectInspector(0, OrcUtils.getOrcTypes(orcSchema));
+  }
+
+  static class Field implements StructField {
+    private final String name;
+    private final ObjectInspector inspector;
+    private final int offset;
+
+    Field(String name, ObjectInspector inspector, int offset) {
+      this.name = name;
+      this.inspector = inspector;
+      this.offset = offset;
+    }
+
+    @Override
+    public String getFieldName() {
+      return name;
+    }
+
+    @Override
+    public ObjectInspector getFieldObjectInspector() {
+      return inspector;
+    }
+
+    @Override
+    public int getFieldID() {
+      return offset;
+    }
+
+    @Override
+    public String getFieldComment() {
+      return null;
+    }
+  }
+
+  static class OrcStructInspector extends SettableStructObjectInspector {
```

Comment:
there's a way to avoid copying this class fully by setting the package name as `org.apache.hadoop.hive.ql.io.orc` given this class is just a package private in Hive's class. Have we tried that ?

Reply:
The issue is that by calling OrcUtils.getOrcTypes, we get org.apache.orc.OrcProto.Type, but OrcStruct.createObjectInspector method only accept org.apache.hadoop.hive.ql.io.orc.OrcProto.Type, that's why I have to rewrite this whole class